exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/json/loader/JsonLoaderImpl.java

```diff
@@ -222,18 +232,41 @@ private JsonLoaderImpl(JsonLoaderBuilder builder) {
     this.rsLoader = builder.rsLoader;
     this.options = builder.options;
     this.errorContext = builder. errorContext;
-    this.rowListener = new TupleListener(this, rsLoader.writer(), builder.providedSchema);
-    this.parser = new JsonStructureParserBuilder()
+    this.fieldFactory = buildFieldFactory(builder);
+    this.parser = buildParser(builder);
+  }
+
+  private JsonStructureParser buildParser(JsonLoaderBuilder builder) {
+    return new JsonStructureParserBuilder()
             .fromStream(builder.stream)
             .fromReader(builder.reader)
             .options(builder.options)
-            .rootListener(rowListener)
+            .parserFactory(new ParserFactory() {
+              @Override
+              public ObjectParser rootParser(JsonStructureParser parser) {
+                return new TupleParser(parser, JsonLoaderImpl.this, rsLoader.writer(), builder.providedSchema);
+              }
+            })
```

Comment:
Please replace it with lambda:
```suggestion
            .parserFactory(parser -> new TupleParser(parser, JsonLoaderImpl.this, rsLoader.writer(), builder.providedSchema))
```